### ISSUE TITLE
Details threshold mask corrupts thumbnails - dt 4.4 Win 10

### ISSUE THREAD

**Issue by europlatus** [darktable-org/darktable#15185]
### Describe the bug

When details threshold is used as a mask in various modules, the thumbnail in the lighttable becomes corrupted. It appears black, white or bright colors.

![image](https://github.com/darktable-org/darktable/assets/89879133/ba768d03-a7b2-44db-b8e3-6881f3ff5a7e)

### Steps to reproduce

1. Open a raw image from the lighttable
2. Apply an edit in a module, e.g. increase exposure 1 stop
3. Open the mask options (either drawn mask or parametric), then adjust the details threshold slider (positive or negative)
4. Return to the lighttable. The thumbnail will be black or otherwise corrupted.

### Expected behavior

The thumbnail should be rendered correctly.

### Where did you install darktable from?

darktable.org

### darktable version

4.4.2

### What OS are you using?

Windows

### What is the version of your OS?

Windows 10

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

GPU: Radeon 7600XT

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

Issue seems to occur with either openCL enabled or disabled. I have tried all the different thumbnail options in the settings, but doesn't seem to make a difference.

Problem may have been introduced with version 4.4 because I didn't have this issue in previous versions.

**Comment by infjb** [comment]
I see the same corruption happening on Windows 11. It began with darktable version 4.4.0.

It rarely also happens in the main editing area after applying blur.

![grafik](https://github.com/darktable-org/darktable/assets/57672052/5bac4ea8-6f9a-4ee6-a35a-e52d61bdfef2)

**Comment by gi-man** [comment]
Lets start with updating the AMD drivers. https://www.amd.com/en/support/graphics/amd-radeon-rx-7000-series/amd-radeon-rx-7600-series/amd-radeon-rx-7600

Install those and reboot windows. Then test dt.


I see this error in your log file. It reads that it falls back to CPU but I dont see it in the log. I know there is a merged PR dealing with demosaic and detail maskl for 4.6. If the GPU driver doesnt address your problem, then testing master might be ideal.

```
    51.4436 [pixelpipe_process_CL]       [thumbnail]    demosaic               (   0/   0) 7710x5196 scale=1.0000 --> (   0/   0)  164x 110 scale=0.0213 IOP_CS_RAW -> IOP_CS_RGB
    51.9243 [write detail mask on GPU]   [thumbnail]                           (   0/   0) 7710x5196 scale=1.0000 
    51.9244 [clip_and_zoom_roi_cl]       [thumbnail]    demosaic               (   0/   0) 7710x5196 scale=1.0000 --> (   0/   0)  164x 110 scale=0.0213 
    51.9248 [interpolation_resample_cl]                                        (   0/   0) 7710x5196 scale=1.0000 --> (   0/   0)  164x 110 scale=0.0213 Error: DT_OPENCL_PROCESS_CL
    52.1932 [clip_and_zoom_roi_cl]                                             (   0/   0) 7710x5196 scale=1.0000 --> (   0/   0)  164x 110 scale=0.0213 did fast cpu fallback
```

**Comment by europlatus** [comment]
I have updated the drivers but the problem remains (there was a typo in my original bug report, I have the 6700XT).
I will try the latest Windows build to see if the problem was fixed in a merged PR. Thanks!

**Comment by europlatus** [comment]
> There have been numerous fixes regarding OpenCL the last three days. So try a **very** fresh one :-)
> 
I've just downloaded and installed the very latest Windows build - darktable-4.5.0+451~g7c8bd1c4d1-win64
From my tests so far, I have not encountered the problem, even when thumbnails are at their smallest and using Lanczos3 (see below), so it may have been fixed. 

>     1. Could you also check with various interpolators?
Seems to only happen with Lanczos3 for scaling interpolator. All other settings (bilnear, bicubic, lanczos2) seem to work fine.
> 
>     2. Is the problem related to thumbnail size?
Yes! It seems to happen when the thumbnails get to a certain small size (19 wide on my 3440x1440 ultrawide monitor). When I increase the size of the thumbnails, they render correctly. 


So, to summarize, the problem only seems to happen with Lanczos3 scaling and only when the thumbnails get to a smaller size.

**Comment by gi-man** [comment]
So, Europlatus, I think we should close this issue since it solved in current master (likely with https://github.com/darktable-org/darktable/pull/15016).

**Comment by jenshannoschwalm** [comment]
Fixed in master, so closing for now
